### PR TITLE
Branch including PDAF-related patching changes

### PR DESCRIPTION
**Pull request by jjokella**
I would like this to become a branch called `tsmp-pdaf-patches` in the HPSCTerrSys owned `CLM3.5` repository. However, I cannot open new branches in that repository, so I opened this PR into `tsmp_patches`.

### CODE REVIEW

**Comment by kvrigor** [comment]
Hi @jjokella! I see no problem merging PDAF patches to `tsmp-patches`. Can you check if `CLM3.5-ParFlow` still builds successfully in [eTSMP](https://github.com/HPSCTerrSys/eTSMP) with this PR? If that works then we can safely merge these changes.

File: src/main/iniTimeConst.F90 [hunk 1]
```diff
@@ -523,7 +541,8 @@ subroutine iniTimeConst
 
       ! Decay factor (m)
 #if (defined CATCHMENT)
-      hkdepth(c) = 1._r8/0.5_r8  ! CPS Lawrence et al. 2011
+      hkdepth(c) = 1._r8/6.0_r8
```
Comment (jjokella) [inline src/main/iniTimeConst.F90]: Discussion of the changed value  for decay factor `hkdepth` needed.

**Comment by jjokella** [comment]
Hi @kvrigor.  Yes, mayeb it would be best to have everything in a single branch if possible. From the current changes, I think the following should be fine
- introduction of `psand`, `pclay`
- changes under preprocessor-flag `WATSAT3D`

The only point of conflict may be the decay factor `hkdepth` that may influence model behavior.

**Comment by jjokella** [comment]
Build of `CLM3.5-ParFlow` was possible with the changes in this PR.

File: src/main/iniTimeConst.F90 (hunk 1 above)
Comment (jjokella) [reply src/main/iniTimeConst.F90]: After internal discussion, I resolved this by setting `hkdepth` to the default value.

**Comment by jjokella** [comment]
@kvrigor  From my side this can now be merged into `tsmp_patches`!

**Comment by kvrigor** [comment]
thanks Johannes!

**Pull request merged**